src/main/java/org/jahia/modules/sitemap/listeners/SitemapSiteListener.java

```diff
@@ -70,10 +70,30 @@ public void onEvent(EventIterator events) {
                 }
                 // Site node removed
                 if (event.getType() == Event.NODE_REMOVED && event.getPath().equals("/sites/" + siteKey)) {
+
                     SitemapJobBuilder builder = new SitemapJobBuilder();
                     builder.createJob = false;
                     jobsToBuild.put(siteKey, builder);
                 }
+                if (event.getType() == Event.PROPERTY_CHANGED && event.getPath().equals("/sites/" + siteKey + "/j:installedModules")) {
+
+                    JCRPropertyWrapper prop = (JCRPropertyWrapper) session.getItem(eventPath);
+                   boolean siteMapNotInstalled = Arrays.stream(prop.getValues()).noneMatch(value -> {
+                        try {
+                            return value.getString().equals("sitemap");
+                        } catch (RepositoryException e) {
+                            throw new RuntimeException(e);
+                        }
+                    });
+                    SitemapJobBuilder builder = new SitemapJobBuilder();
+                    if (siteMapNotInstalled) {
+                        builder.createJob = false;
+                       builder.deleteSitemap = true;
+                    } else {
+                        builder.createJob = true;
```

Comment:
I am not sure we want to create the job at the time the module is enabled on a site as we are missing all the config information